app/actions/gfwclient.py

```diff
@@ -492,14 +551,30 @@ async def get_gfw_integrated_alerts(self, *, geostore_id: str,date_range: Tuple[
         
         return [IntegratedAlert.parse_obj(alert) for alert in alerts] if alerts else []
         
+    @backoff.on_exception(backoff.expo, (httpx.TimeoutException, httpx.HTTPStatusError), max_tries=3, on_backoff=backoff_hdlr)
+    async def get_nasa_viirs_fire_alerts(self, *, geostore_id: str,date_range: Tuple[datetime, datetime], semaphore: asyncio.Semaphore = None):
+
+        async with semaphore:
```

Comment:
I imagine this semaphore will limit requests within the context of a single instance, and depending on the cloudrun settings the integration service could scale out to multiple instances. So we just need to consider that and set `GFW_DATASET_QUERY_CONCURRENCY` but also `max_instances` properly. If we are setting `max_instances=1` for GFW then we can manage concurrency just by adjusting `GFW_DATASET_QUERY_CONCURRENCY` and that'd be fine.

Reply:
Yes, it's intended to limit concurrent requests to the dataset query endpoint, and I do indeed calibrate the maximum concurrent requests with this semaphore in mind. I'll add a note in the code for future readers.